src/Simplex/Messaging/Agent.hs

```diff
@@ -189,20 +189,20 @@ deleteUser :: AgentErrorMonad m => AgentClient -> UserId -> Bool -> m ()
 deleteUser c = withAgentEnv c .: deleteUser' c
 
 -- | Create SMP agent connection (NEW command) asynchronously, synchronous response is new connection id
-createConnectionAsync :: forall m c. (AgentErrorMonad m, ConnectionModeI c) => AgentClient -> UserId -> ACorrId -> Bool -> SConnectionMode c -> m ConnId
-createConnectionAsync c userId corrId enableNtfs cMode = withAgentEnv c $ newConnAsync c userId corrId enableNtfs cMode
+createConnectionAsync :: forall m c. (AgentErrorMonad m, ConnectionModeI c) => AgentClient -> UserId -> ACorrId -> Bool -> SubscriptionMode -> SConnectionMode c -> m ConnId
+createConnectionAsync c = withAgentEnv c .::. newConnAsync c
```

Comment:
not a good idea, given that there is a boolean and now we don't know what it does.

Reply:
also, given that notifications mode and SConnectionMode are permanent, and SubscriptionMode is session-scoped, it logically should be the last parameter everywhere